From ce195bb7ad26b51102ad1f5d54594cfcbb92acba Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:25:30 -0400
Subject: [tools] perf/trace: Prepare the strarray scnprintf method for reuse

Message-id: <1408462094-14194-162-git-send-email-jolsa@redhat.com>
Patchwork-id: 88131
O-Subject: [PATCH RHEL7.1 BZ1131394 161/325] perf trace: Prepare the strarray scnprintf method for reuse
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 975b7c2f40d431da4fdb46dd8b9d90c129eecc12
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Tue Oct 8 17:17:43 2013 -0300

description
===========
Right now when an index passed to that method has no string associated
it'll print the index as a decimal number, prepare it so that we can use
it to print it in hex as well, for ioctls, for instance.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index e1fa7cc..0de348d 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -60,18 +60,25 @@ struct strarray {
  .entries = array, \
 }
 
-static size_t syscall_arg__scnprintf_strarray(char *bf, size_t size,
-           struct syscall_arg *arg)
+static size_t __syscall_arg__scnprintf_strarray(char *bf, size_t size,
+      const char *intfmt,
+             struct syscall_arg *arg)
 {
  struct strarray *sa = arg->parm;
  int idx = arg->val - sa->offset;
 
  if (idx < 0 || idx >= sa->nr_entries)
-  return scnprintf(bf, size, "%d", arg->val);
+  return scnprintf(bf, size, intfmt, arg->val);
 
  return scnprintf(bf, size, "%s", sa->entries[idx]);
 }
 
+static size_t syscall_arg__scnprintf_strarray(char *bf, size_t size,
+           struct syscall_arg *arg)
+{
+ return __syscall_arg__scnprintf_strarray(bf, size, "%d", arg);
+}
+
 #define SCA_STRARRAY syscall_arg__scnprintf_strarray
 
 static size_t syscall_arg__scnprintf_fd(char *bf, size_t size,
-- 
1.7.1